tests/naive.py

```diff
@@ -1388,55 +1436,81 @@ def aampdist_snippets(
     )
 
 
-def prescrump(T_A, m, T_B, s, exclusion_zone=None):
+def prescrump(T_A, m, T_B, s, exclusion_zone=None, k=1):
     dist_matrix = distance_matrix(T_A, T_B, m)
 
     l = T_A.shape[0] - m + 1  # matrix profile length
     w = T_B.shape[0] - m + 1  # distance profile length
 
-    P = np.empty(l)
-    I = np.empty(l, dtype=np.int64)
-    P[:] = np.inf
-    I[:] = -1
+    P = np.full((l, k), np.inf, dtype=np.float64)
+    I = np.full((l, k), -1, dtype=np.int64)
 
     for i in np.random.permutation(range(0, l, s)):
         distance_profile = dist_matrix[i]
         if exclusion_zone is not None:
             apply_exclusion_zone(distance_profile, i, exclusion_zone, np.inf)
 
-            # only for self-join
-            mask = distance_profile < P
-            P[mask] = distance_profile[mask]
-            I[mask] = i
-
-        I[i] = np.argmin(distance_profile)
-        P[i] = distance_profile[I[i]]
-        if P[i] == np.inf:
-            I[i] = -1
-        else:
-            j = I[i]
-            for k in range(1, min(s, l - i, w - j)):
-                d = dist_matrix[i + k, j + k]
-                if d < P[i + k]:
-                    P[i + k] = d
-                    I[i + k] = j + k
-                if exclusion_zone is not None and d < P[j + k]:
-                    P[j + k] = d
-                    I[j + k] = i + k
+        nn_idx = np.argmin(distance_profile)
+        if distance_profile[nn_idx] < P[i, -1] and nn_idx not in I[i]:
+            pos = np.searchsorted(P[i], distance_profile[nn_idx], side="right")
+            P[i] = np.insert(P[i], pos, distance_profile[nn_idx])[:-1]
+            I[i] = np.insert(I[i], pos, nn_idx)[:-1]
+
+        if P[i, 0] == np.inf:
+            I[i, 0] = -1
+            continue
+
+        j = nn_idx
+        for g in range(1, min(s, l - i, w - j)):
+            d = dist_matrix[i + g, j + g]
+            if d < P[i + g, -1] and (j + g) not in I[i + g]:
+                pos = np.searchsorted(P[i + g], d, side="right")
+                # Do NOT optimize the `condition` in the following if statement
```

Comment:
move the comment and put it just before the if condition. Also, check out its performant version.

Reply:
Is this what you need to do or you've already done? I can't tell if you need me to do look at something here?